### PR TITLE
Issue #7594: Update doc for UncommentedMain

### PR DESCRIPTION
**Pull request by AmrDeveloper**
Fixes Issue: #7594 

uncommentedmain

Output of default example:

    $ cat config2.xml

    <?xml version="1.0"?>
    <!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://checkstyle.sourceforge.net/dtds/configuration_1_3.dtd">
    <module name="Checker">
        <module name="TreeWalker">
           <module name="UncommentedMain"/>
        </module>
    </module>

    $ cat Test2.java

    public class Game{
       public static void main(String[] args){}  // violation
    }

    public class Main{
       public static void main(String[] args){}  // violation
    }

    public class Launch{
       //public static void main(String[] args){} // OK
    }

    $ java -jar checkstyle-8.29-all.jar -c config2.xml Test2.java
    Starting audit...
    [ERROR] D:\OpenSource\TestCommit\Test2.java:2: Uncommented main method found. [UncommentedMain]
    [ERROR] D:\OpenSource\TestCommit\Test2.java:6: Uncommented main method found. [UncommentedMain]
    Audit done.
    Checkstyle ends with 2 errors.

Output of excludedClasses example:

    $ cat config2.xml

    <?xml version="1.0"?>
    <!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://checkstyle.sourceforge.net/dtds/configuration_1_3.dtd">
    <module name="Checker">
        <module name="TreeWalker">
           <module name="UncommentedMain">
              <property name="excludedClasses" value="\.Main$"/>
            </module>
        </module>
    </module>

    $ cat Test2.java

    public class Game{
       public static void main(String[] args){}  // violation
    }

    public class Main{
       public static void main(String[] args){}  // OK
    }

    public class Launch{
       //public static void main(String[] args){} // OK
    }

    $ java -jar checkstyle-8.29-all.jar -c config2.xml Test2.java
    Starting audit...
    [ERROR] D:\OpenSource\TestCommit\Test2.java:2: Uncommented main method found. [UncommentedMain]
    Audit done.
    Checkstyle ends with 1 errors.

### CODE REVIEW

File: src/xdocs/config_misc.xml [hunk 1]
```diff
@@ -2197,7 +2197,22 @@ messages_home_fr_CA_UNIX.properties
         <source>
 &lt;module name=&quot;UncommentedMain&quot;/&gt;
         </source>
+        <p>
+          Example:
+        </p>
```
Comment (baratali) [inline src/xdocs/config_misc.xml]: please change it to: `<p>Example:</p>`

File: src/xdocs/config_misc.xml [hunk 2]
```diff
@@ -2207,6 +2222,22 @@ messages_home_fr_CA_UNIX.properties
   &lt;property name=&quot;excludedClasses&quot; value=&quot;\.Main$&quot;/&gt;
 &lt;/module&gt;
         </source>
+        <p>
+          Example:
+        </p>
```
Comment (baratali) [inline src/xdocs/config_misc.xml]: please change it to: `<p>Example:</p>`

**Comment by baratali** [comment]
Other changes are ok with me.
@romani , @rnveach , please review. Do you have preference between `<pre>` and `<source>` in xml docs? I just saw commits in master using `<pre>` (https://github.com/checkstyle/checkstyle/commit/c7ae67e353b25e263dd5d2f852cff220bd43dd20#diff-7b7a98a340dc4db79fa5029ae03bd945R2140) and `<source>` (https://github.com/checkstyle/checkstyle/commit/3d799201c5e1fcbd04b9c146971faebc45786dd3#diff-7b7a98a340dc4db79fa5029ae03bd945R3833)

File: src/xdocs/config_misc.xml (hunk 1 above)
Comment (romani) [reply src/xdocs/config_misc.xml]: @AmrDeveloper , please do update as requested by @baratali .

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/UncommentedMainCheck.java [hunk 3]
```diff
@@ -52,6 +52,20 @@
  * <pre>
  * &lt;module name=&quot;UncommentedMain&quot;/&gt;
  * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * public class Game {
+ *    public static void main(String[] args){}   // violation
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/UncommentedMainCheck.java]: please add example of method that is named "main" but is not public or not static or parameters are not array of string.

Please add example of modern style "String... args":
https://stackoverflow.com/a/8756966/1015848

**Comment by AmrDeveloper** [comment]
@romani @baratali Done

**Pull request merged**